services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -0,0 +1,56 @@
+package cromwell.pipeline.service
+import java.nio.file.Path
+
+import com.typesafe.config.{ Config, ConfigFactory }
+import cromwell.pipeline.datastorage.dto.{ Project, ProjectFile, Version }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class GitLabProjectVersioning(httpClient: HttpClient)(
+  implicit executionContext: ExecutionContext
+) extends ProjectVersioning[VersioningException] {
+  import GitLabConfig._
+
+  override def updateFile(project: Project, projectFile: ProjectFile): AsyncResult[String] = ???
+  override def updateFiles(project: Project, projectFiles: ProjectFiles): AsyncResult[List[String]] = ???
+
+  override def createRepository(project: Project): AsyncResult[Project] = {
+    def responseFuture = httpClient.post(URL + "projects", createRepositoryParams(project), TOKEN, "")
+    if (!project.active)
+      Future.failed(VersioningException("Could not create a repository for not an active project."))
+    else
+      responseFuture.flatMap(
+        resp =>
+          Future.successful(
+            if (resp.status != 201)
```

Comment:
I think you need a constant for this

Reply:
Agree. There would be more methods and maybe more different response statuses. I will fix it.